app/controllers/campaign_pages_controller.rb

```diff
@@ -23,6 +23,14 @@ def create
     i = 0
     widgets.each do |widget_type_name, widget_data|
       widget_type_id = widget_data.delete('widget_type')
+
+
+      if defined? widget_data['checkboxes']['{cb_number}']
```

Comment:
I think this is testing the method definition of Hash#[] which I imagine isn't what you intended. 

```ruby
widget_data['checkboxes']['{cb_number}'].present?
```